src/System.Management.Automation/namespaces/FileSystemContentStream.cs

```diff
@@ -725,7 +725,18 @@ private bool ReadDelimited(bool waitChanges, ArrayList blocks, bool readBackward
 
             // We've reached the end of file or end of line.
             if (content.Length > 0)
-                blocks.Add(content.ToString());
+            {
+                // Add the block read to the ouptut array list, trimming a trailing delimiter, if present.
+                // Note: If -Tail was specified, we get here in the course of 2 distinct passes:
+                //  - Once while reading backward simply to determine the appropriate *start position* for later forward reading, ignoring the content of the blocks read (in reverse).
+                //  - Then again during forward reading, for regular output processing; it is only then that trimming the delimiter is necessary.
+                //    (Trimming it during backward reading would not only be unnecessary, but could interfere with determining the correct start position.)
+                blocks.Add(
+                    !readBackward && content.ToString().EndsWith(actualDelimiter, StringComparison.Ordinal)
+                        ? content.ToString().Substring(0, content.Length - actualDelimiter.Length)
+                        : content.ToString()
```

Comment:
Here we evaluate `content.ToString()` twice - maybe do this one before `block.Add()`.